daemon/src/address_map.rs

```diff
@@ -49,3 +90,24 @@ impl<'a, K, A> Disconnected<'a, K, A> {
         };
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use xtra::Context;
+
+    #[test]
+    fn gc_removes_address() {
+        let (addr, _) = Context::<Dummy>::new(None);
+        let mut map = AddressMap::default();
+        map.insert("foo", addr.clone());
+
```

Comment:
it would be beneficial to have a `"bar" address there too, to show that it doesn't stop all addresses